src/frontend/components/split-pane/split-pane.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * <split-pane> component.
+ *
+ * This may seem like a mess, but there is no other way to implement resizable panes.
+ */
+
+import {
+  Component,
+  ElementRef,
+  ViewChild,
+} from '@angular/core';
+
+// This value matches the behaviour in the Chrome Dev Tools
+const MIN_PANE_WIDTH = 26;
+
+const DEFAULT_SECONDARY_WIDTH = 384;
+
+@Component({
+  selector: 'split-pane',
+  templateUrl: '/src/frontend/components/split-pane/split-pane.html'
+})
+export default class SplitPane {
+  @ViewChild('wrapper') wrapperElement : ElementRef;
+  @ViewChild('resizer') resizerElement : ElementRef;
+  @ViewChild('overlay') overlayElement : ElementRef;
+  @ViewChild('primary') primaryElement : ElementRef;
+  @ViewChild('secondary') secondaryElement : ElementRef;
+
+  // TODO: store the initial secondary pane width in a preference.
+  secondaryWidth = DEFAULT_SECONDARY_WIDTH;
+
+  // State for resizing
+  mouseX : number;
+  bounds : ClientRect;
+  guardMouseMoved;
```

Comment:
Unfortunately, TypeScript complains when this is typed `Function`, possibly due to a typing bug. It seems to have a rule for methods named `addEventListener` which requires the use of a higher level object with a `handleEventMethod()` on it. I think it is applying that rule incorrectly, and in any case the type it thinks addEventListener wants is _not_ the type that addEventListener actually wants; so I have to leave this untyped.